File: Id1/LangSolutions/julia/src/main.jl
#!/usr/local/bin/julia
include("MyModule.jl")
using Printf

function main()
    #Values ​​of challenge 1 of the Euler project
    up_bound = 1000
    multiple1 = 3 
    multiple2 = 5
    #Variables for time measurement
    total_time = 0.0
    num_iterations = 100
    #Warm-up execution, prepares the system to perform iterative execution and better measure the execution time
    result = MyModule.solve(up_bound, multiple1, multiple2)
    #Measuring the average time gives a better result than measuring a single execution.
    for i in 1:num_iterations
        _, time_elap, _, _, _ = @timed MyModule.solve(up_bound, multiple1, multiple2)
        total_time += time_elap
    end

    average_time = total_time / num_iterations

    println("\tResult: ",result)
    println("\tTime elapsed: ", @sprintf("%.3e", average_time), "s")
end

main()

File: Id1/LangSolutions/rust/src/main.rs
mod module;
use std::time::Instant;

/// The main code is practically the same for all the challenges of the Euler Project,
/// the only changes are those referring to the variables and value of the challenge
/// to be solved.
/// # Note
/// For information on the logic see 'mylib.py' in the module folder
fn main() {
    //Values ​​of challenge 1 of the Euler project.
    let up_bound = 1000;
    let multiple1 = 3;
    let multiple2 = 5;
    //Warm-up execution, prepares the system to perform iterative execution and better measure the execution time
    let result = module::solve(up_bound - 1, multiple1, multiple2);
    //Variables for time measurement
    let mut total_duration = 0f64;
    let num_executions = 100;
    for _ in 0..num_executions {
        let start = Instant::now(); //Start of time measurement
        module::solve(up_bound - 1, multiple1, multiple2);
        let duration = start.elapsed(); //End of time measurement
        total_duration += duration.as_secs_f64();
    }
    let average_duration = total_duration / num_executions as f64; //converts the result to float, by default it comes in ..μs format
    println!("\tResult: {}", result);
    println!("\tAverage time: {:.3e}s", average_duration);    
}